test/framework/filetools.py

```diff
@@ -1201,10 +1205,34 @@ def test_apply_regex_substitutions(self):
 
         # passing empty list of substitions is a no-op
         ft.write_file(testfile, testtxt)
-        ft.apply_regex_substitutions(testfile, [])
+        ft.apply_regex_substitutions(testfile, [], on_missing_match=run.IGNORE)
         new_testtxt = ft.read_file(testfile)
         self.assertEqual(new_testtxt, testtxt)
 
+        # Check handling of on_missing_match
+        ft.write_file(testfile, testtxt)
+        regex_subs_no_match = [('Not there', 'Not used')]
+        error_pat = 'Nothing found to replace in %s' % testfile
+        # Error
+        self.assertErrorRegex(EasyBuildError, error_pat, ft.apply_regex_substitutions, testfile, regex_subs_no_match,
+                              on_missing_match=run.ERROR)
+
+        fancylogger.logToFile(self.logfile)
+
+        # Warn
+        ft.apply_regex_substitutions(testfile, regex_subs_no_match, on_missing_match=run.WARN)
+        with open(self.logfile, 'r') as f:
+            logtxt = f.read()
+        self.assertTrue('WARNING ' + error_pat in logtxt)
+
+        # Ignore
+        ft.apply_regex_substitutions(testfile, regex_subs_no_match, on_missing_match=run.IGNORE)
+        with open(self.logfile, 'r') as f:
+            logtxt = f.read()
+        self.assertTrue('INFO ' + error_pat in logtxt)
+
+        fancylogger.logToFile(self.logfile, enable=False)
```

Comment:
This should (also) be in `tearDown`, to make sure it's always done, also if the tests fails somewhere in the middle?

Reply:
`tearDown` is tricky. It would be run for all tests, which might have side effects. I added a contextmanager which solves that better.